checkST2110.js

```diff
@@ -1073,108 +1305,317 @@ const test_21_82_2 = (sdp, params) => {
+
+// Test ST 2110-22 Section 7.2 Test 1 - Test all required parameters are present
+const test_22_72_1 = (sdp, params) => {
+  let [mtParams, errors] = extractMTParams(sdp, { checkDups: true });
+  for (let stream of mtParams) {
+    let keys = Object.keys(stream);
+    for (let param of mustHaves22) {
+      if (keys.indexOf(param) < 0) {
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, required parameter '${param}' is missing, as per SMPTE ST2110-22 Section 7.2`));
+      }
+    }
+    for (let param of mustHaves9134) {
+      if (keys.indexOf(param) < 0) {
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, required parameter '${param}' is missing, as per RFC-9134 Section 7.1 of RFC-9134`));
+      }
+    }
+  }
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: Test ST2110-22 Section 7.2 RFC-9134 Section 7.1 Test 1 - All required parameters are present');
+  }
+  return errors;
+};
+
+// Test ST 2110-22 Section 73 -  Check for mandatory bitrate attribute in correct format
+const test_22_73_1 = (sdp, params) => {
+  let errors = [];
+  let lines = splitLines(sdp);
+  let bandwidthPresent = false;
+  for (let x = 0; x < lines.length; x++) {
+    if (lines[x].startsWith('b=')) {
+      let bandwidthMatch = lines[x].match(bandwidthPattern);
+      if (bandwidthMatch == null) {
+        errors.push(new Error(`Line ${x + 1}: In 'Bandwidth must be of the form 'b=<bwtype>:<bandwidth>' as per SMPTE ST2110-22 Section 7.3.`));
+        continue;
+      }
+      if (bandwidthMatch[1] != 'AS')
+        errors.push(new Error(`Line ${x + 1}: In 'b=<bwtype>:<bandwidth>' bwtype must be 'AS' as per SMPTE ST2110-22 Section 7.3.`));
+      if (Number.isInteger(+bandwidthMatch[2]) == false)
+        errors.push(new Error(`Line ${x + 1}}: In 'b=<bwtype>:<bandwidth>' bandwidth must be specified as an integer as per SMPTE ST2110-22 Section 7.3.`));
+      bandwidthPresent = true;
+    }
+  }
+  if (!bandwidthPresent) {
+    errors.push(new Error('Required attribute \'b=<bwtype>:<bandwidth>\' is missing, as per SMPTE ST2110-22 Section 7.3.'));
+  }
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: Test ST 2110-22 Section 7.3 -  Check for mandatory bitrate attribute');
   }
   return errors;
 };
 
+// Test ST 2110-22 Section 74 Test 1-  Check that framerate is specified by one of accepted methods
+const test_22_74_1 = (sdp, params) => {
+  let lines = splitLines(sdp);
+  let framerateAttributePresent = false;
+  let framerateParameterPresent = false;
+  // First check if exactframerate is specified as a parameter of fmtp
+  let [mtParams, errors] = extractMTParams(sdp, params);
+  if(mtParams.length == 1) {
+    if(mtParams[0].exactframerate != null) {
+      framerateParameterPresent = true;
+    }
+  }else {
+    errors.push(new Error('SMPTE-2110-22 SDP File should specify a single video stream or a duplicate for SMPTE2110-7'));
```

Comment:
ST 2022-7 intended, I assume?

```suggestion
    errors.push(new Error('ST 2110-22 SDP File should specify a single video stream or a duplicate for ST 2022-7'));
```